LsifDotnet/Program.cs

```diff
@@ -84,6 +88,7 @@ private static void ConfigureHost(IHostBuilder host)
 
         host.ConfigureServices((_, collection) => collection.AddSingleton(_ => CreateWorkspace())
             .AddSingleton<IdentifierCollectorFactory>()
+            .AddSingleton<MatcherFactory>()
```

Comment:
If we just want to have a singleton of macher, we can just use `.AddSingleton<Matcher>()`